Programs/brlapi_client.c

```diff
@@ -1659,12 +1660,33 @@ static int getControllingTty(void)
 
 #ifdef linux
   {
-    FILE *stream;
+    int fd;
+    if ((fd = open("/proc/self/stat", O_RDONLY, 0)) >= 0) {
+      char buf[256];
+      size_t read_bytes = read(fd, buf, sizeof(buf) - 1);
+      close(fd);
+      /* Append null at the end of buffer to ensure that read doesn't
+       * overflows.*/
+      buf[255] = '\0';
+
+      /* The line format is "$pid ($comm) $state ...
+       * where comm may contain space and `)` as part of its name, ex:
+       * 12345 (someproc) S 123 456) R ...
+       * So to process valid identified after comm, we should skip comm by
+       * getting the 1st ) from end.
+       */
+      char *ptr = buf + read_bytes - 1;
+      while (buf < ptr && *ptr != ')') {
```

Comment:
I'd say rather use `ptr >= buf`, that'll look more idiomatic

Reply:
Update to `ptr > buf`, since if we do `>=` check and for some reason the file is not formatted correctly, we may end up at buf - 1, and the next if checks for *(buf - 1), which is something that we should probably avoid. If we are at `buf + 1` and we still haven't found ')', then even if the prev char is ')' that would still be invalid format, so it should be safe to avoid.

Let me know if this sounds right.